### PR TITLE
Add typeahead field

### PR DESCRIPTION
**Pull request by brunojdo**
It's about the issue #67

---

This change is [Reviewable](https://reviewable.io/reviews/danielfireman/isumm/68)

### CODE REVIEW

File: static/js/typeahead.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**
+  Typeahead function - This function enables the typeahead input
+  and writes the investment id in #hiddenInputElement to form submit
+**/
+$(document).ready(function(){
+  $('#investment-field .typeahead').typeahead({
+    hint: true,
```
Comment (danielfireman) [inline static/js/typeahead.js]: Make this a constant? (Maybe JS engine does this, but I am not sure)

File: static/js/typeahead.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    highlight: true,
+    minLength: 1
+  },
+  {
+    name: 'investments',
+    source: substringMatcher(investments)
```
Comment (danielfireman) [inline static/js/typeahead.js]: Where are you getting investments variable from?

**Comment by danielfireman** [comment]
Pretty awesome change!

File: static/js/typeahead.js (hunk 2 above)
Comment (brunojdo) [inline static/js/typeahead.js]: On the app.template.html look at the <script></script> block in the head of html.

Comment (danielfireman) [inline static/js/typeahead.js]: This wiring is very fragile. I have an idea how to make this slightly better. Lets chat.

**Comment by danielfireman** [comment]
As discussed online, app.html and ops.html should have something like this:

$document.ready(typeaheadOn('fieldId','hiddenFieldId', investments))

and typeahead function returns a function that uses the parameters as global variables. This reduces a lot the scope of these guys.

**Comment by danielfireman** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Comment by brunojdo** [comment]
Review status: 1 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

_[static/js/typeahead.js, line 13 [r1]](https://reviewable.io:443/reviews/danielfireman/isumm/68#-KMVdtPYjkK8XiU9Qzq9-r1-13:-KMWWnD2AczcV_vwQxkA:-499146009) ([raw file](https://github.com/danielfireman/isumm/blob/699dc186b16504e53b29ff94c2901d9e2ecab92c/static/js/typeahead.js#L13)):_
<details><summary>

<i>Previously, danielfireman (Daniel Fireman) wrote…</i></summary>



> This wiring is very fragile. I have an idea how to make this slightly better. Lets chat.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Comment by danielfireman** [comment]
:lgtm:

---

Review status: 1 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68#-:-KMWXBHNbRJlARqokeFH:1417088721)_

**Comment by brunojdo** [comment]
Review status: 1 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

_[static/js/typeahead.js, line 7 [r1]](https://reviewable.io:443/reviews/danielfireman/isumm/68#-KMVdtPYjkK8XiU9Qzq9-r1-7:-KMWXZP4XvPreuCS01pg:-499146009) ([raw file](https://github.com/danielfireman/isumm/blob/699dc186b16504e53b29ff94c2901d9e2ecab92c/static/js/typeahead.js#L7)):_
<details><summary>

<i>Previously, danielfireman (Daniel Fireman) wrote…</i></summary>



> Make this a constant? (Maybe JS engine does this, but I am not sure)
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Comment by danielfireman** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Comment by danielfireman** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[static/js/typeahead.js, line 7 [r1]](https://reviewable.io:443/reviews/danielfireman/isumm/68#-KMVdtPYjkK8XiU9Qzq9-r1-7:-KMWYNBoLiaJEY2B_FCf:5173697) ([raw file](https://github.com/danielfireman/isumm/blob/699dc186b16504e53b29ff94c2901d9e2ecab92c/static/js/typeahead.js#L7)):_
<details><summary>

<i>Previously, brunodias20 (Bruno Dias) wrote…</i></summary>



> Done.
> </details>
> Ok

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Comment by danielfireman** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[static/js/typeahead.js, line 13 [r1]](https://reviewable.io:443/reviews/danielfireman/isumm/68#-KMVdtPYjkK8XiU9Qzq9-r1-13:-KMWYRThrOCMe5iktDDj:5173697) ([raw file](https://github.com/danielfireman/isumm/blob/699dc186b16504e53b29ff94c2901d9e2ecab92c/static/js/typeahead.js#L13)):_
<details><summary>

<i>Previously, brunodias20 (Bruno Dias) wrote…</i></summary>



> Done.
> </details>
> Ok

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/danielfireman/isumm/68)_

**Pull request merged**